views/menus/stav.js

```diff
@@ -4,67 +4,199 @@ import React from 'react'
 import {
   StyleSheet,
   View,
-  NetInfo,
-  WebView,
+  ListView,
+  Text,
+  ActivityIndicator,
 } from 'react-native'
 
-const styles = StyleSheet.create({
-  container: {
-    flex: 1,
-  },
-  webView: {
-    flex: 1,
-  },
-})
+import FoodItem from './foodItem'
+
+import moment from 'moment-timezone'
+const CENTRAL_TZ = 'America/Winnipeg'
 
-const URL = 'http://stolaf.cafebonappetit.com/cafe/stav-hall#Lunch'
-// const DEFAULT_URL = 'http://legacy.cafebonappetit.com/print-menu/cafe/261/menu/112732/days/today/pgbrks/0/'
+import DietaryFilters from './dietaryFilters'
+const dietaryFilters = DietaryFilters()
+
+import buildingHours from '../../data/building-hours.json'
 
 export default class StavMenuView extends React.Component {
-  state = {
-    isConnected: true,
-    url: URL,
-    scalesPageToFit: true,
+  constructor(props){
+    super(props)
+    let getSectionData = (dataBlob, sectionID) => {
+      return dataBlob[sectionID]
+    }
+
+    let getRowData = (dataBlob, sectionID, rowID) => {
+      return dataBlob[rowID]
+    }
+
+    this.state = {
+      loaded: false,
+      dataSource: new ListView.DataSource({
+        getSectionData: getSectionData,
+        getRowData: getRowData,
+        rowHasChanged: (row1, row2) => row1 !== row2,
+        sectionHeaderHasChanged: (s1, s2) => s1 !== s2,
+      }),
+    }
   }
 
-  componentWillMount() {
-    NetInfo.isConnected.fetch().then(isConnected => {
-      this._onConnectivityChange(isConnected)
-    })
-    NetInfo.isConnected.addEventListener('change', this._onConnectivityChange.bind(this))
+  componentDidMount(){
+    this.fetchData()
   }
 
-  componentWillUnmount() {
-    NetInfo.isConnected.removeEventListener('change', this._onConnectivityChange.bind(this))
+  getTodaysLunchHours(currentDay){
+    for (let i = 0; i < buildingHours.length; i++){
+      if (buildingHours[i].name=='Stav Hall Lunch'){
+        return buildingHours[i].times.hours[currentDay][1]
+      }
+    }
   }
 
-  _onConnectivityChange(isConnected: boolean) {
-    this.setState({isConnected})
+  whichMeal(){
+    let current = moment().tz(CENTRAL_TZ)
+    let currentTime = current.format('H:mm')
+    let lunchClose = this.getTodaysLunchHours(current.format('ddd'))
+
+    if (currentTime < lunchClose){
+      return 1
+    } else {
+      return 2
+    }
   }
 
-  onShouldStartLoadWithRequest() {
-    return true
+  buildStation(station, items){
+    let stationItems = []
+    for (let i = 0; i < station.items.length; i++){
+      let food = items[station.items[i]]
+      if (food.special == 1){
+        let temp = {
+          name: food.label,
+          dietary: food.cor_icon,
+        }
+
+        stationItems.push(temp)
+      }
+    }
+    return stationItems
   }
 
-  onNavigationStateChange(navState: {url: string}) {
-    this.setState({url: navState.url})
+
+  fetchData () {
```

Comment:
You'll want to make this a bound instance function with `fetchData = () => {` – it's an ES2017(?) change that binds the function to the class instance. We can use it because the react native processor turns it on for us. 

Without that, the call to `this.setState` – well, I'd think that it would never work, but it might? I think it's a bug if it does.